lib/bmp280.ex

```diff
@@ -120,45 +121,48 @@ defmodule BMP280 do
 
     {:ok, transport} = Transport.open(bus_name, bus_address)
 
-    state = %{
-      transport: transport,
-      calibration: nil,
-      sea_level_pa: Keyword.get(args, :sea_level_pa, @sea_level_pa),
-      sensor_type: nil
-    }
+    state =
+      %{
+        transport: transport,
+        calibration: nil,
+        sea_level_pa: Keyword.get(args, :sea_level_pa, @sea_level_pa),
+        sensor_type: nil,
+        last_measurement: nil
+      }
+      |> query_sensor()
+      |> init_sensor()
 
     {:ok, state, {:continue, :continue}}
   end
 
   @impl GenServer
   def handle_continue(:continue, state) do
-    new_state =
-      state
-      |> query_sensor()
-      |> init_sensor()
+    send(self(), :schedule_measurement)
 
-    {:noreply, new_state}
+    {:noreply, state}
   end
 
   @impl GenServer
-  def handle_call(:measure, _from, state) do
-    rc =
-      case read_raw_samples(state.transport, state.sensor_type) do
-        {:ok, raw} ->
-          {:ok,
-           Calc.raw_to_measurement(
-             state.calibration,
-             state.sea_level_pa,
-             raw
-           )}
+  def handle_info(:schedule_measurement, state) do
+    Process.send_after(self(), :schedule_measurement, @polling_interval)
+
+    case read_raw_samples(state.transport, state.sensor_type) do
+      {:ok, raw} ->
+        measurement =
+          Calc.raw_to_measurement(
+            state.calibration,
+            state.sea_level_pa,
+            raw
+          )
 
-        error ->
-          error
-      end
+        {:noreply, %{state | last_measurement: {:ok, measurement}}}
 
-    {:reply, rc, state}
+      {:error, _} = error ->
```

Comment:
What do you think about adding a timestamp to the measurement, keeping the most recent measurement around, and logging errors? It could be fancier on the errors, but the idea would be that forcing the altitude would almost always work and that `measure` wouldn't fail due to a hiccup with the sensor.

Reply:
Sounds good. What time format would you like? In Web development, ISO8601 is commonly used. I think it is most developer friendly.

In my personal project, I use Timex library, but `DateTime.utc_now` might be good enough and it does not need any dependency. 

```
iex(5)> "#{DateTime.utc_now}"
"2021-02-23 01:54:02.706164Z"

iex(6)>   Timex.format!(Timex.now(), "{ISO:Extended}")
"2021-02-23T01:54:50.015915+00:00"
```

Reply:
I'll work on in in a separate PR since we will probably need to change `Measurement` struct as well.

Reply:
Good question. I always go with `System.monotonic_time(:millisecond)` and save it to a variable named `timestamp_ms` (or similar, but with the _ms suffix). The reason is that my use of timestamps is mostly to answer the question of how much time has passed since the last notification. Monotonic time is the easiest way of making this calculation since it's just subtraction.

Converting monotonic time to wall clock time isn't too hard and I usually don't do that until I need to show it to the user.

One thing with Nerves is that the wall clock time jumps around especially at boot when the network comes up and network time synchronization happens. We try to minimize the jumps with some tricks, but it's not possible to fix unless the board has a real-time clock on it.

Reply:
I changed my mind and added `timestamp_ms` to `Measurement` struct in this PR. It was simple enough to be part of this PR. Next, improvement for error handling.

Reply:
I tested current code on my target device. https://github.com/fhunleth/bmp280/pull/16/commits/08d4089b68f342eb8069cf9c849c648eac8f5d05

```elixir
iex(3)> BMP280.measure(sensor)
%BMP280.Measurement{
  altitude_m: 100.08783874848216,
  dew_point_c: 5.552282714860013,
  humidity_rh: 19.02895307060607,
  pressure_pa: 100307.4539495844,
  temperature_c: 32.03974440819584,
  timestamp_ms: 65390
}
iex(4)> BMP280.measure(sensor)
%BMP280.Measurement{
  altitude_m: 100.12688563124946,
  dew_point_c: 5.552646084676659,
  humidity_rh: 19.023982308536954,
  pressure_pa: 100306.98860420777,
  temperature_c: 32.044804466329516,
  timestamp_ms: 67392
}
```